master/internal/api_project.go

```diff
@@ -364,29 +364,37 @@ func (a *apiServer) GetProjectsByUserActivity(
 	}
 
 	err = db.Bun().NewSelect().Model(p).NewRaw(`
-	WITH p as (
-		SELECT  pr.*, activity_time FROM projects pr 
-		JOIN activity a ON pr.id = a.entity_id WHERE a.user_id = ? ORDER BY a.activity_time DESC LIMIT ?
-	),
-	pe AS (
-	  SELECT project_id, state, start_time
-	  FROM experiments
-	)
-	SELECT w.name as workspace_name, u.username, p.id, p.name, 
-	p.workspace_id, p.description, p.immutable, p.notes, p.user_id,
-			'WORKSPACE_STATE_' || p.state AS state, p.error_message,
-	  SUM(case when pe.project_id = p.id then 1 else 0 end) AS num_experiments,
-	  SUM(case when pe.project_id = p.id 
-		AND pe.state = 'ACTIVE' then 1 else 0 end) AS num_active_experiments,
-	  MAX(case when pe.project_id = p.id then pe.start_time else NULL end) 
-	  AS last_experiment_started_at
-	FROM p
-	  LEFT JOIN users as u ON u.id = p.user_id
-	  LEFT JOIN workspaces AS w on w.id = p.workspace_id
-	  LEFT JOIN pe on pe.project_id = p.id 
-	GROUP BY p.user_id, p.id, p.name, p.workspace_id, p.description, 
-	p.immutable, p.notes, p.state, p.error_message,  u.username, w.name, p.activity_time
-	ORDER BY p.activity_time DESC;`, curUser.ID, limit).
+	SELECT
+		w.name AS workspace_name,
+		u.username,
+		p.id,
+		p.name,
+		p.archived,
+		p.workspace_id,
+		p.description,
+		p.immutable,
+		p.notes,
+		p.user_id,
+		'WORKSPACE_STATE_' || p.state AS state,
+		p.error_message,
+		COUNT(*) FILTER (WHERE e.project_id = p.id) AS num_experiments,
```

Comment:
nice finding. i just followed the previous implementation.
I'll test it without the filter tomorrow

Reply:
looks like we need the filter. the result changed without it

Reply:
I'd recommend figuring out the why's as a way to better understand the systems involved.

in this case, I believe the projects with no experiments will have a row will null experiment columns. So `count(*) == 1` because it's one row, but `count ... WHERE` or `count(distinct e.id)` will be 0 because e.id is null.